src/modules/layout/containers/HeaderContainer.js

```diff
@@ -47,13 +47,11 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
   const cityCode = state.cityContent?.city
   const categoriesAvailable = state.cityContent?.searchRoute !== null
 
-  const cityModel = cities.find(city => city.code === cityCode)
```

Comment:
You probably could have instantiated `cityModel` here already:
```js
const cityModel = route ? cities.find(city => city.code === route.cityCode) : null
```
in order to pass it in the first step already when other things are still loading...